### PR TITLE
Asc 60 space counters

### PR DESCRIPTION
**Pull request by mfrancisc**
Add SpaceCounter in toolchainstatus CRD

### CODE REVIEW

**Review by xcoulon — APPROVED** [review]

**Review by xcoulon — CHANGES_REQUESTED** [review]
did you regenerate the "deepcopy", Open API and CRD files files using `make generate`?

**Review by xcoulon — APPROVED** [review]

**Comment by xcoulon** [comment]
> yes, see: https://github.com/mfrancisc/host-operator/blob/eb933f58e7123e539a54c945cc6a2b4d7c792a94/config/crd/bases/toolchain.dev.openshift.com_toolchainstatuses.yaml#L493
> 
> Next time I'll do it in separate PR's.

yes, indeed, I saw that file afterwards 🤦‍♂️ 
I'm surprised there is no change in the deepcopy and OpenAPI files, though 🤔

**Comment by mfrancisc** [comment]
right, I was checking as well, but I think it's because SpaceCounter it's an int - primitive type - (as UserAccountCount for example), and not a struct.

**Comment by mfrancisc** [comment]
but I see zz_generated.openapi.go has reference to userAccountCount so I guess there is something missing. let me try again.

**Comment by xcoulon** [comment]
I just ran `make generate` and got the following changes in this repo:
```
diff --git api/v1alpha1/zz_generated.openapi.go api/v1alpha1/zz_generated.openapi.go
index d1c79c6..e2d2716 100644
--- api/v1alpha1/zz_generated.openapi.go
+++ api/v1alpha1/zz_generated.openapi.go
@@ -1107,6 +1107,13 @@ func schema_codeready_toolchain_api_api_v1alpha1_Member(ref common.ReferenceCall
                                                        Format:      "int32",
                                                },
                                        },
+                                       "spaceCount": {
+                                               SchemaProps: spec.SchemaProps{
+                                                       Description: "Number of Spaces created within the member cluster",
+                                                       Type:        []string{"integer"},
+                                                       Format:      "int32",
+                                               },
+                                       },
                                        "memberStatus": {
                                                SchemaProps: spec.SchemaProps{
                                                        Description: "The array of member status objects",
```

**Comment by mfrancisc** [comment]
right, maybe it's a mismatch in the generation tools.

I have controller-gen v0.8.0:
```
➜  api git:(ASC-60_space_counters) ./bin/controller-gen --version
Version: v0.8.0
```
but there is not way to get `openapi-gen` version apparently ( I don't see a `version` flag/command).

I remember having issues with the make targets not downloading the binaries for `controller-gen` and `opeanpi-gen` , so I had to download them and place them in `./bin` directory my self.



I also have go1.18.6 but that should work I guess ...

**Comment by xcoulon** [comment]
> right, maybe it's a mismatch in the generation tools.
> 
> I have controller-gen v0.8.0:
> 
> ```
> ➜  api git:(ASC-60_space_counters) ./bin/controller-gen --version
> Version: v0.8.0
> ```
> 
> but there is no way to get `openapi-gen` version apparently ( I don't see a `version` flag/command).
> 
> I remember having issues with the make targets not downloading the binaries for `controller-gen` and `opeanpi-gen` , so I had to download them and place them in `./bin` directory my self.
> 
> I also have go1.18.6 but that should work I guess ...

I have the same version v0.8.0 for `./bin/controller-gen` and I'm running on Go 1.17 but I don't think this should make a difference 🤔

**Review by xcoulon — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]

File: api/v1alpha1/zz_generated.openapi.go [hunk 1]
```diff
@@ -1107,6 +1107,13 @@ func schema_codeready_toolchain_api_api_v1alpha1_Member(ref common.ReferenceCall
 							Format:      "int32",
 						},
 					},
+					"spaceCount": {
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply api/v1alpha1/zz_generated.openapi.go:1110]: @mfrancisc so what was the reason that it wasn't generated for the first time?

[the remark this answers is not in the document]
Comment (mfrancisc) [reply api/v1alpha1/zz_generated.openapi.go:1110]: my local `api` folder was located under `/Users/fmuntean/go/src/github.com/api` and not `/Users/fmuntean/go/src/github.com/codeready-toolchain/api`.

The `ORG` part was missing and expected by the `--output-package` [here](https://github.com/codeready-toolchain/api/blob/7a7f4eb137dbef6ddd1cdfbdac8dadad9a4fdebc/make/generate.mk#L27), which is actually an alias for _destination folder_ under GOPATH. 

My mistake! @xcoulon helped me figure this out!

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply api/v1alpha1/zz_generated.openapi.go:1110]: Ok, I see.
Thanks for the description of the problem :+1:
